snippets/csharp/System.Net.Sockets/Socket/Receive/sendgeneric.cs

```diff
@@ -73,15 +71,15 @@ public static int syncSendAndReceive(string host, int port)
         public static void SendCallback(IAsyncResult ar)
         {
             allDone.Set();
-            Socket s = (Socket) ar.AsyncState;
-            s.EndSend(ar);
+            Socket? s = (Socket?)ar.AsyncState;
+            s?.EndSend(ar);
```

Comment:
I might consider this pattern instead:

```suggestion
            if (ar.AsyncState is Socket s)
            {
                s.EndSend(ar);
            }
```